components/specification/transforms/util/write-transforms.py

```diff
@@ -0,0 +1,189 @@
+
+def shortest_path(from_schema, to_schema, min_quality):
+    paths = [[from_schema]]
+    while True:
+        # any paths lead to the goal?
+        for path in paths:
+            if path[-1] == to_schema:  # yes, goal path found
+                return path
+
+        # extend all the current paths
+        next_paths = []
+        for path in paths:
+            curr_step = path[-1]
+            for next_step in transforms[curr_step]:
+                if quality(curr_step, next_step) >= min_quality and \
+                        next_step not in path:
+                    next_path = list(path)
+                    next_path.append(next_step)
+                    next_paths.append(next_path)
+
+        if not next_paths:  # no more transforms to try
+            return None
+        paths = next_paths
+
+
+# the style of transform file names
+
+name_pattern = compile('^(.+)\-to\-(.+)\.xsl$')
+
+
+# scan the current directory to determine the schemas and transforms
+
+def load_transforms():
+    global transforms
+    global schemas
+    seen = set()
+
+    for name in listdir('.'):
+        match = name_pattern.match(name)
+        if match:
+            from_schema = match.group(1)
+            to_schema = match.group(2)
+
+            transforms[from_schema].append(to_schema)
+            seen.add(from_schema)
+            seen.add(to_schema)
+
+    schemas = list(seen)
+    schemas.sort(reverse=True)
+
+
+# note of the best sequence of transforms among the schemas
+
+best_paths = {}
+
+
+# determine the best sequences of transforms among the schemas
+
+load_transforms()
+
+for from_schema in schemas:
+    for to_schema in schemas:
+        if from_schema != to_schema:
+            for min_quality in range(4, 0, -1):
+                path = shortest_path(from_schema, to_schema, min_quality)
+                if path:
+                    best_paths[(from_schema, to_schema)] = (path, min_quality)
+                    break
+            if not best_paths[(from_schema, to_schema)]:
+                raise Exception(
+                    'no path from ' + from_schema + ' to ' + to_schema)
+
+
+# name the quality levels of transforms
+
+qualities = ['poor', 'fair', 'good', 'excellent']
+
+
+# print in XML a sequence of transforms to the given schema
+
+def print_path(to_schema):
+    (path, min_quality) = best_paths[(from_schema, to_schema)]
+    print '\t\t\t<target schema="' + to_schema + \
+        '" quality="' + qualities[min_quality - 1] + '">'
+    while len(path) > 1:
+        print '\t\t\t\t<transform file="' + \
+            path[0] + '-to-' + path[1] + '.xsl"/>'
+        path = path[1:]
+    print '\t\t\t</target>'
+
+
+# print in XML all the transforms among the schemas
+
+print """<?xml version = "1.0" encoding = "UTF-8"?>
+<!--
+	#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+	#
+	# Copyright (C) 2012-2014 Open Microscopy Environment
```

Comment:
I figured that we're not really adding anything new to the transforms file outside the regions space so on this branch there's probably not much to protect. @rleigh-dundee: Once this PR is merged I could open a PR merging BF into regions and updating this to 2015, and once OMERO 5.2.0 is tagged I could open a PR merging that into regions, if you like? Basically, I'm happy to help, but also happy to keep clear of conflicts with your work.